Analytics/importers/scheduler_new.py

```diff
@@ -0,0 +1,133 @@
+import os, sys
+from datetime import datetime, timedelta
+import time
+import importlib
+import logging
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+import sqlalchemy
+from sqlalchemy.orm import sessionmaker, scoped_session
+from apscheduler.schedulers.background import BackgroundScheduler
+from apscheduler.executors.pool import ProcessPoolExecutor, ThreadPoolExecutor
+
+import settings
+
+logging.basicConfig(level='INFO', filename='importers.log', filemode='a')
+logger = logging.getLogger(__name__)
+
+
+class Scheduler(object):
+    """ Schedule the execution importer tasks """
+
+    def __init__(self):
+        """ Create SQLAlchemy session and launch APScheduler """
+
+        self.engine = sqlalchemy.create_engine(settings.DB_URI)
+        _session = scoped_session(sessionmaker(bind=self.engine))
+        self.session = _session
+
+        executor = {
+            'default': ThreadPoolExecutor(10),
+            'processpool': ProcessPoolExecutor(5)
+        }
+        self.scheduler = BackgroundScheduler(executors=executor)
+        self.scheduler.start()
+
+    def get_apis(self) -> list:
+        """
+        Retrieve API details stored in the database
+        :return: a list containing the details of APIs
+        """
+
+        _api = self.session.execute('select * from api')
+        apis = []
+        for api in _api.fetchall():
+            a = API(name=api[1], url=api[2], api_key=api[3],
+                    api_class=api[4],
+                    refresh_time=api[5], token_expiry=api[6],
+                    timestamp=api[7])
+            apis.append(a)
+        return apis
+
+    def fetch_data(self, class_name: str, api_name: str):
+        """
+        Import sensor data from the specified API endpoint
+        :param class_name: name of the class that implements the
+        corresponding importer
+        :param api_name: identifying name stored in the database for the
+        respective importer
+        """
+
+        from app import create_app
```

Comment:
Why is this not in the imports at the top of the file?

Reply:
The fetch_data function was implemented by an FCC developer so I can't exactly answer this question. But when I do try to place the imports at the top of the file, the import process fails